core/src/main/java/pl/agh/edu/time_command/TimeCommandExecutor.java

```diff
@@ -1,34 +1,28 @@
 package pl.agh.edu.time_command;
 
 import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.PriorityQueue;
 
 public class TimeCommandExecutor {
-    private static TimeCommandExecutor instance;
-    private final Map<LocalDateTime, List<TimeCommand>> commands;
+    private static final TimeCommandExecutor instance = new TimeCommandExecutor();
+    private final PriorityQueue<TimeCommand> commands;
 
     private TimeCommandExecutor(){
-        this.commands = new HashMap<>();
+        this.commands = new PriorityQueue<>();
     }
 
     public static TimeCommandExecutor getInstance() {
-        if(instance == null) return new TimeCommandExecutor();
         return instance;
     }
 
-    public void addCommand(LocalDateTime dateTime, TimeCommand timeCommand) {
-        if (!commands.containsKey(dateTime)) {
-            commands.put(dateTime, new ArrayList<>());
-        }
-        commands.get(dateTime).add(timeCommand);
+    public void addCommand(TimeCommand timeCommand) {
```

Comment:
jak mam dwie komendy: A i B, obie na 12:00 i obie są repeating every day to mamy pewność, że zawsze zostaną wywołane w kolejności: najpierw A a potem B? Chodzi mi o to czy dodawanie do priorityQueue zawsze wstawi nową komende jako ostatnią z posiadających ten sam czas?

Reply:
wcześniej było to zupełnie losowe teraz dodałem pole z czasem stworzenia i jest po tym porównywalne więc mamy pewność że te które stworzyliśmy pierwsze będą pierwsze wykonywane

Reply:
Nie podoba mi się że uzależniamy się tutaj od czasu rzeczywistego. Może w takim wypadku dorobimy jakiś priorytet do TimeCommanda wtedy te komendy, które muszą być przed innymi bedą miały wysoki priorytet. Plus rozwiązał by się problem z tym, że kolejność wywołania komend jest zależna od kolejności wstawiania ich do executora. No ale to może byc zrobione w nowym tasku.